src/App.tsx

```diff
@@ -26,70 +28,118 @@ type ReorderOptions = {
   isReversed: boolean,
 };
 
-// Use this function in the render method to prepare goods
 export function getReorderedGoods(
   goods: string[],
   { sortType, isReversed }: ReorderOptions,
-) {
-  // To avoid the original array mutation
+): string[] {
   const visibleGoods = [...goods];
 
-  // Sort and reverse goods if needed
-  // eslint-disable-next-line no-console
-  console.log(sortType, isReversed);
+  if (sortType !== SortType.NONE) {
+    visibleGoods.sort((good1, good2) => {
+      switch (sortType) {
+        case SortType.LENGTH:
+          return good1.length - good2.length;
+        case SortType.ALPHABET:
+          return good1.localeCompare(good2);
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
 
   return visibleGoods;
 }
 
-// DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
-
-      <ul>
-        <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
-        </ul>
-      </ul>
-    </div>
-  );
+type State = {
+  isReversed: boolean,
+  sortType: SortType,
 };
+
+export class App extends Component<{}, State> {
+  state: Readonly<State> = {
+    isReversed: false,
+    sortType: SortType.NONE,
+  };
+
+  sortAlphabetically = () => {
+    this.setState({ sortType: SortType.ALPHABET });
+  };
+
+  SortByLength = () => {
+    this.setState({ sortType: SortType.LENGTH });
+  };
+
+  reverse = () => {
+    this.setState(state => ({
+      isReversed: !state.isReversed,
+    }));
+  };
+
+  reset = () => {
+    this.setState({
+      isReversed: false,
+      sortType: SortType.NONE,
+    });
+  };
+
+  render() {
+    const { isReversed, sortType } = this.state;
+    const sorted = isReversed || sortType !== SortType.NONE;
```

Comment:
Naming conventions are important in programming because they make code more readable and easier to understand. Here are some common naming conventions for boolean variables and functions:

**Boolean Variables:**
- Start with "is", "has", "should", or "does" to indicate that the variable represents a true or false condition.
- Use descriptive names that accurately convey the meaning of the variable.
- Use `camelCase` for multi-word names (e.g. "isCompleted" instead of "is_completed").

Examples:
```
isLoggedin
hasPermission
shouldShowDialog
doesExist
```

**Functions:**
- Start with a verb that describes the action the function performs.
- Use descriptive names that accurately convey the purpose of the function.
- Use `camelCase` for multi-word names.

Examples:
```
handleClick
validateForm
renderListItems
deleteAccount
```

These naming conventions help to make code more readable and self-explanatory, which is especially important when working on larger projects with other developers.